pkg/operation/botanist/component/kubestatemetrics/resources.go

```diff
@@ -274,6 +274,7 @@ func (k *kubeStateMetrics) reconcileDeployment(
 					},
 				},
 			}},
+			PriorityClassName: v1beta1constants.PriorityClassNameSeedSystem600,
```

Comment:
This will also set this priority when the `kube-state-metrics` is deployed for the shoot cluster. However, in this case we want to have `gardener-system-100`.